### PR TITLE
revives: Move typetraits.`$` to system. Fixes #5827

### LINKED ISSUE #5827: Nim beginner's feedback: "echo type(1)" does not work
I've recently introduced Nim to a number of colleagues (mostly with a C++ and Go background), and I want to share some of the feedback I got.

Surprise: you cannot print the type of an expression
```
echo 1
echo type(1)    # does not compile
```

This seems easy to fix with the following patch (not sure if that is correct), but I wonder why something like this not enabled by default given that the `type()` proc is exported from system.nim.
```
diff --git a/lib/system.nim b/lib/system.nim
index 9b41253..ce41373 100644
--- a/lib/system.nim
+++ b/lib/system.nim
@@ -168,6 +168,8 @@ proc `type`*(x: untyped): typeDesc {.magic: "TypeOf", noSideEffect, compileTime.
   ## Builtin 'type' operator for accessing the type of an expression.
   ## Cannot be overloaded.
   discard
+from typetraits import nil
+proc `$`*(t: typedesc): string = typetraits.name(t)
 
 proc `not` *(x: bool): bool {.magic: "Not", noSideEffect.}
   ## Boolean not; returns true iff ``x == false``.
```

### PR DESCRIPTION
**Pull request by timotheecour**
* fixes #5827
* revives https://github.com/nim-lang/Nim/pull/7100 by addressing comments and fixing test failures
/cc @dom96 I kept your 1st commit so your name is still there

### CODE REVIEW

**Comment by timotheecour** [comment]
btw: on my todo list for this pr is using a template `name` instead of proc in typetraits, will try today

**Comment by Araq** [comment]
> @Araq seeing you close PRs without a comment explaining why is really demoralising.

I did explain why. Nobody bothered to make the CIs green for a long time.

**Comment by dom96** [comment]
> I did explain why. Nobody bothered to make the CIs green for a long time.

No you haven't. Your last comment was 10 days ago and you didn't explicitly say "I am closing this because CIs aren't green".

File: lib/system.nim [hunk 1]
```diff
@@ -4431,3 +4431,19 @@ when defined(genode):
       componentConstructHook(env)
         # Perform application initialization
         # and return to thread entrypoint.
+
+proc name(t: typedesc): string {.magic: "TypeTrait".}
+  ## pending https://github.com/nim-lang/Nim/issues/10078, nest inside
+  ## `$`*(t: typedesc)
+
+template `$`*(t: typedesc): string =
```
Comment (alaviss) [inline lib/system.nim]: Why can't this be a `proc`? Doesn't the `typedesc` parameter already make it a compile time evaluation?

**Comment by timotheecour** [comment]
ping @dom96

File: lib/system.nim (hunk 1 above)
Comment (Araq) [reply lib/system.nim]: This should be a ``proc`` and ``semmagic.nim`` line 139 needs to be updated to read like ``of "name", "$"``.

Comment (timotheecour) [reply lib/system.nim]: done

**Comment by timotheecour** [comment]
PTAL

**Pull request merged**